src/cli/llm/infer_type_name.rs

```diff
@@ -1,15 +1,16 @@
 use std::collections::HashMap;
 
 use genai::chat::{ChatMessage, ChatRequest, ChatResponse};
+use indexmap::{indexmap, IndexMap};
 use serde::{Deserialize, Serialize};
 use serde_json::json;
 
-use super::{Error, Result, Wizard};
+use super::{Error, Result, TypeUsageIndex, Wizard};
 use crate::core::config::Config;
 use crate::core::Mustache;
 
-pub struct InferTypeName {
-    wizard: Wizard<Question, Answer>,
+pub struct InferTypeName<'a> {
+    wizard: Wizard<Question<'a>, Answer>,
```

Comment:
I don't think we need to keep a lifetime parameter.

Reply:
actually we do, Question expects lifetime parameter.
```rust
#[derive(Clone, Serialize)]
struct Question<'a> {
    references: IndexMap<&'a str, u32>,
    fields: Vec<(&'a str, &'a str)>,
}
```